integration/autoupdate/tools/updater_tsh_test.go

```diff
@@ -329,6 +329,56 @@ func TestMigratedUpdateNotReExec(t *testing.T) {
 	matchVersion(t, string(out), testVersions[0])
 }
 
+// TestUpdateConfigReExecPath verifies that after update and re-execution we preserve the original
+// path of the executed binary, because `tsh config` is referencing this path for execution.
+func TestUpdateConfigReExecPath(t *testing.T) {
+	t.Setenv(types.HomeEnvVar, t.TempDir())
+	ctx := context.Background()
+
+	rootServer, _, _ := bootstrapTestServer(t)
+	setupManagedUpdates(t, rootServer.GetAuthServer(), autoupdate.ToolsUpdateModeEnabled, testVersions[1])
+
+	proxyAddr, err := rootServer.ProxyWebAddr()
+	require.NoError(t, err)
+
+	// Fetch compiled test binary with updater logic and install to $TELEPORT_HOME.
+	updater := tools.NewUpdater(
+		toolsDir,
+		testVersions[0],
+		tools.WithBaseURL(baseURL),
+	)
+	err = updater.Update(ctx, testVersions[0])
+	require.NoError(t, err)
+
+	tshPath, err := updater.ToolPath("tsh", testVersions[0])
+	require.NoError(t, err)
+
+	cmd := exec.CommandContext(ctx, tshPath,
+		"login", "--proxy", proxyAddr.String(), "--insecure", "--user", "alice", "--auth", constants.LocalConnector)
+	cmd.Env = os.Environ()
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	require.NoError(t, cmd.Run())
+
+	// Execute version command again with setting the new version which must
+	// trigger re-execution of the same command after downloading requested version.
+	cmd = exec.CommandContext(ctx, tshPath, "version")
+	cmd.Env = os.Environ()
+	out, err := cmd.Output()
+	require.NoError(t, err)
+
+	matchVersion(t, string(out), testVersions[1])
+
+	cmd = exec.CommandContext(ctx, tshPath, "config", "--proxy", proxyAddr.String(), "--insecure")
+	cmd.Env = os.Environ()
+	out, err = cmd.Output()
+	require.NoError(t, err)
+
+	// Ensure that original path of tsh before update is present in configuration generated by
+	// config command.
+	require.Contains(t, string(out), tshPath)
```

Comment:
```suggestion
	require.Contains(t, string(out), `ProxyCommand "` + tshPath + `" `)
```

by the way, `tsh config` is just broken if tsh is in a path that contains quotes or newlines, isn't it 😬